src/trackedit/internal/trackeditactionscontroller.cpp

```diff
@@ -578,6 +582,18 @@ void TrackeditActionsController::silenceAudioSelection()
     pushProjectHistoryTrackSilenceState(selectedStartTime, selectedEndTime);
 }
 
+void TrackeditActionsController::renderClipPitchAndSpeed(const muse::actions::ActionData& args)
+{
+    trackedit::ClipKey clipKey = args.arg<trackedit::ClipKey>(0);
+    if (!clipKey.isValid()) {
+        return;
+    }
+
+    trackeditInteraction()->renderClipPitchAndSpeed(clipKey);
+
+    pushProjectHistoryRenderClipStretchingState();
```

Comment:
This should be part of the logic of the trackeditInteraction()->renderClipPitchAndSpeed method, i.e. if we call trackeditInteraction()->renderClipPitchAndSpeed from some place, then the logic associated with history should also be executed